### ISSUE TITLE
pause the autoscroll

### ISSUE THREAD

**Issue by qidizi** [termux/termux-app#1242]
**Feature description**
something will cost a long time,
sometime i want to find out what it say,
but i can not .
either press ctrl+c to break working for read,
or wait for work is finish,but may be some output is loss.

**Reference implementation**
mac osx terminal
when you scrollup,autoscroll will stoping,
when scrollend it will keeping scrolling.

**Comment by AbdullahM0hamed** [comment]
Until something is done about this, you can pipe output to less as a temporary solution. Output isn't lost, and you can follow along at your own pace.

**Comment by ZimbiX** [comment]
Very annoying indeed. I'd have thought there'd be a setting for this, but I can't find one.

That `less` idea is interesting, but I imagine I'd still often come across situations where it's not until I've already commenced a long command with output (without `less`) that I realise I need to see some scrollback.

I don't imagine this would be difficult to implement, but I could be wrong.

**Comment by ZimbiX** [comment]
Ah, I see this is a duplicate of https://github.com/termux/termux-app/issues/684 where it's suggested to use `tmux`. I guess it's about time I learn how to use that

**Comment by buffermet** [comment]
It's about time termux has this basic functionality.

**Comment by SDRausty** [comment]
Try `emacs` for this.  It's been around for decades.

**Comment by buffermet** [comment]
@SDRausty can you please elaborate? AFAIK emacs is an alternative to Vim.

@ZimbiX `less` breaks most escaped output, even when `-r` or `-R` flags are set.

I know Android has the most painful excuse for word-wrapping. They call it [breakStrategy](https://developer.android.com/reference/android/widget/TextView#attr_android:breakStrategy) and its accepted values make absolutely no sense (balanced, high_quality, simple). Web developers know you either want break-none, break-word or break-all.

Since termux does behave as "break-all", it means they probably devised their own grid system, which means it should be easy enough to lock onto a certain row. Not sure why it's taking so long to implement this because having to inspect streaming output whilst fighting against the automatic scrolling is disturbing to say the least.

so tmux it is...

**Comment by SDRausty** [comment]
> emacs 

search string: run command in emacs

I have not done this in quite some time:  This was...  Compiling the Linux kernel on i286 and more recent processors.  Kinescope screen time and buffer were issues.  Emacs performs well.  

alt search string: run command in vim

@qidizi are you familiar with?

search string: redirect output bash site:tldp.org

**Comment by qidizi** [comment]
> > AFAIK emacs is
> 
> search string: `run command in emacs`
> 
> I have not done this in quite some time: This was... Compiling the Linux kernel on i386, including more recent processors, the box, keyboard and the 3d box screen. The screen resembled a portable TV btw. Kinescope screen time and the small screen buffer were issues. 40 lines of screen, then the screen output is gone. Emacs performs well...
> 
> alt search string: `run command in vim`
> 
> @qidizi are you familiar with this topic at tldp.org?
> 
> search string: `redirect output bash site:tldp.org`
> 
> **The ability to pause autoscrolling, including searching autoscrolling is available through tools such as emacs, redirecting output (a native shell ability), vim, [stream editors](https://wiki.termux.com/wiki/Stream_Editors), etc...**
> 
> How would a user access a native ability to pause autoscrolling in Termux? A Termux meta key in the [extra keys row](https://wiki.termux.com/wiki/Touch_Keyboard)? A meta key popup like: long screen press > More... > Help which shows [our help page](https://wiki.termux.com)?
> 
> Once autoscrolling is paused, will the user be able to manually scroll the screen as desired until manually unpausing autoscrolling?


`ping 127.0.0.1`   
output overflow and autoscrolling  

when swipe up n pixel paused  
when swipe down to last line keep autoscrolling.

**Comment by SDRausty** [comment]
@qidizi is long tap until the context menu shows when the screen is scrolling what you are looking for?  When the menu shows up, the scrolling pauses.

**Comment by qidizi** [comment]
> @qidizi is long tap until the context menu shows when the screen is scrolling what you are looking for? When the menu shows up, the scrolling pauses.

Yes, it will pause. But it's not a friendly operation. For example, if I want to copy another line, I have to click on the screen to cancel the menu, but it's scrolling out of the visible area. The ping operation only adds 1 line per second, and maybe you can easily hold down the line you want, what if it's a cat log? Maybe you'll terminate the operation and use grep, so what if the add-on line is extremely fast and can't be used multiple times? So I think it's very friendly to roll up n px like the terminal of mac and pull down to the end line and resume scrolling.

**Comment by eda-abec** [comment]
There is a function for this in most terminals. I found that you can also enable it in Termux like this:

`stty ixon` (you will probably want to place this to some init script)

and then, to pause:
`ctrl-S`

and to resume
`ctrl-Q`

Just tried it with the ping command. And its clean, no workaround. :)

Refernece: https://unix.stackexchange.com/a/12108

**Comment by buffermet** [comment]
Pausing the terminal is not the same. If there is a large output stream, resuming output will make you lose your scroll position and you will have to scroll all the way up to continue reading.

This is an essential feature that should have been implemented years ago 🤷‍♂️ workarounds it is.

**Comment by eda-abec** [comment]
I really agree with you and it annoys me, too. I just thought I could share this way in case it suits someones needs. And its deeply integrated, as I said. :smiley:

**Comment by SDRausty** [comment]
Why are all my comments minimized?  The answers thst I have received from @agnostic-apollo and @Grimler91:

"I am spamming," by @Grimler91 

"I am throwing tantrums," by @agnostic-apollo 

Please pardon my comment that both of these answers are intimidation and perversion of work asked for, then bashed.  If Termux is asking/pleading for assistance, then outright bashing/deleting/hiding/ignoring and minimizing valid contibutions that truely work in Termux without rooting Android, the main motivation seems to be, "if it doesn't work, root your Android," logic at all costs to our end users, and Termux app development itself.  Kind of like Putin's plan, sacrifice Russia for the entire postwar world.

**Comment by SDRausty** [comment]
Why are all my comments minimized in this issue, and all across the board at https://github.com/termux suddenly minimized? The answers that I have received from @agnostic-apollo and @Grimler91:

"I am spamming," by @Grimler91 

"I am throwing tantrums," by @agnostic-apollo 

Please pardon my comment that both of these answers are intimidation and perversion of work asked for, then bashed.  If Termux is asking/pleading for assistance, then outright bashing/deleting/hiding/ignoring and minimizing valid contibutions that truely work in Termux without rooting Android, the main motivation seems to be, "if it doesn't work, root your Android," logic at all costs to our end users, and Termux app development itself.  Kind of like Putin's plan to sacrifice Russia for the entire postwar world.

**Comment by wmcbtech30** [comment]
> Why are all my comments minimized in this issue, and all across the board at https://github.com/termux suddenly minimized? The answers that I have received thus far from @agnostic-apollo and @Grimler91:

> https://github.com/termux/termux-app/issues/686

> https://github.com/termux/termux-packages/issues/202#issuecomment-1072904804_

You were posting off-topic/non-sense comments on every issue we find. you keep sponsoring your TermuxArch project when if it's about porting new packages for Termux itself, it's a nice project but it would be pointless adding new packages if we rely on running programs in Arch instead of Termux natively (why not we turn this project into Arch ecosystem). and you also post irrelevant comments in most issues that hasn't been found helpful at all or even out of topic when we're focusing on a specific issue of a problem. you've been struggling to stay on-topic associated with the issue

> If Termux is asking/pleading for assistance, then Termux is outright bashing/deleting/hiding/ignoring and minimizing valid contibutions that truely work in Termux without rooting Android, then the main motivation seems to be, "if it doesn't work, root your Android," logic at all costs to our end users and Termux app and [Termux apps](https://github.com/termux/termux-api-package/pulls?q=is%3Apr+author%3ASDRausty) and to https://github.com/termux/termux-packages/issues/8758.

We don't force users to root their devices just to use Termux. there is a specific reason why packages cannot be ported. contributions and suggestions are welcome but you don't be seem to be helpful at all

And also this is not the right place to complain such stuff. use appropriate place (e.g. discussions) for your complains and suggestions that you've been trying to address than posting it here

I hope you realize on what you've been doing and why we need to minimize your comments because we find it irrelevant and unhelpful. if you keep throwing tantrums and not realizing your mistakes then we would definitely ban you from this organization

**Comment by SDRausty** [comment]
>off-topic/non-sense comments

Examples with links please, I am not asking for more spam wall bashing.  Please aquant yourself with TermuxArch's goals.  One of which is to show what can be easily ported into Termux as it does work, as opposed to does not work today, if ever.

>use appropriate place (e.g. discussions) for your complains and suggestions 

How, as my account has been repeatedly banned at Gitter and through other disscussion channels as well by Termux?

**Comment by wmcbtech30** [comment]
> Examples with links please, I am not asking for more spam wall bashing. Please aquant yourself with TermuxArch's goals. One of which is to show what can be easily ported into Termux as it does work, as opposed to does not work today, if ever.

https://github.com/termux/termux-app/issues/686#issuecomment-680977125
https://github.com/termux/termux-packages/issues/202#issuecomment-1072665323
https://github.com/termux/termux-packages/issues/202#issuecomment-1072665323

> How, as my account has been repeatedly banned at Gitter and through other disscussion channels as well by Termux?

https://github.com/termux/termux-app/discussions

**Comment by SDRausty** [comment]
>right place

Op @qidizi 
>Unable to pause autoscrolling
>press ctrl+c to break 

How has minimizing these comments in this thread contributing to Termuxcs smooth and rapid development?

>Try `emacs` for this.  It's been around for decades.  

>>> AFAIK emacs is 

>search string: ` run command in emacs `

>I have not done this in quite some time:  This was...  Compiling the Linux kernel on i386, including more recent processors, the box, keyboard and the 3d box screen.  The screen resembled a portable TV btw.  Kinescope screen time and the small screen buffer were issues.  40 lines of screen, then the screen output is gone.  Emacs performs well...

>alt search string: ` run command in vim `

>@qidizi are you familiar with this topic at tldp.org?

>search string: ` redirect output bash site:tldp.org ` 

>**The ability to pause autoscrolling, including searching autoscrolling is available through tools such as emacs, redirecting output (a native shell ability), vim, [stream editors](https://wiki.termux.com/wiki/Stream_Editors), etc...**

>How would a user access a native ability to pause autoscrolling in Termux?  A Termux meta key in the [extra keys row](https://wiki.termux.com/wiki/Touch_Keyboard)?  A meta key popup like: long screen press > More... > Help which shows [our help page](https://wiki.termux.com)?

>Once autoscrolling is paused, will the user be able to manually scroll the screen as desired until manually unpausing autoscrolling? 

>@qidizi is long tap until the context menu shows when the screen is scrolling what you are looking for?  When the menu shows up, the scrolling pauses.

@wmcbtech30 said
>we find it irrelevant and unhelpful. if you keep throwing tantrums

I am not surprised @fornwall is absent to ask for help.  How were my comments in this thread irrelevant and unhelpful?

@qidizi said
>Unable to pause autoscrolling
>press ctrl+c to break working for read,

Op's question answered is irrelevant and unhelpful?  How?

**Comment by wmcbtech30** [comment]
we wouldn't minimize your comments if you know exactly what "Contributing" really does. you are struggling elaborating your post. your comments doesn't really makes sense at all. even if you contribute for years, you probably didn't realize that your comments are actually hidden and realize it today

**Comment by SDRausty** [comment]
>And this is where you spam your project three times:

I hope that you are aware that as far as we knew then in lste 2017, thst it was impossible for `povray` to work in Termux as far as anyone knew.  It took many device hours and much difficult and hard work to announce a breakthrough in Termux and for Termux!  How is sharing newly uncovered milestones after milestones just more spam @wmcbtech30?

**Comment by wmcbtech30** [comment]
Considering that `proot` adds overhead. it wouldn't be optimal to run `povray` in arch. especially if people have very low end device that would stressing the CPU and could impact on performance. you do know the point of package request issues is to port it for Android specifically and not focus on another thing. people would automatically install a distribution if they find out that package is not present at termux repos but yours seem to keep advertising your arch project

**Comment by SDRausty** [comment]
> Examples with links please

> Please aquant yourself with TermuxArch's goals. One of which is to show what can be easily ported into Termux as it does work, as opposed to does not work today, if ever.

**Comment by wmcbtech30** [comment]
considering it works in arch. it was compiled with different toolchain, while some packages introduce compilation problems and some device has narrow disk space and indecent specifications
